include/extractor/guidance/intersection.hpp

```diff
@@ -71,7 +76,10 @@ struct ConnectedRoad final : public TurnOperation
 // small helper function to print the content of a connected road
 std::string toString(const ConnectedRoad &road);
 
-struct Intersection final : public std::vector<ConnectedRoad>
+// `ConnectedRoads` is a relative view of an intersection by an incoming edge.
+// `ConnectedRoads` are streets at an intersection ordered from from sharp right counter-clockwise to
+// sharp left where `connected_roads[0]` is _always_ a u-turn
+struct ConnectedRoads final : public std::vector<ConnectedRoad>
```

Comment:
Why the change in the API here? This will break _all_ users throughout osrm-backend.

Also semantically it it an intersection represented as ordered roads.